Lib/test/test_inspect/test_inspect.py

```diff
@@ -4622,6 +4622,122 @@ class D2(D1):
 
         self.assertEqual(inspect.signature(D2), inspect.signature(D1))
 
+    def test_signature_skip_bound_arg_method(self):
+        def decorator(func):
+            @functools.wraps(func)  # set `__wrapper__` attribute
+            def wrapper(*args, **kwargs):
+                return func(*args, **kwargs)
+            return wrapper
+
+        class My:
+            def method(self, arg: int) -> None: ...
+            @decorator
+            def decorated(self, arg: int) -> None: ...
+            @classmethod
+            def cl(cls, arg2: str) -> None: ...
+            @staticmethod
+            def st(arg1: str) -> bool: ...
+
+        for follow_wrapped in (True, False):
+            unbound_sigs = {
+                My().method: '(self, arg: int) -> None',
+                My().decorated: (
+                    '(self, arg: int) -> None'
+                    if follow_wrapped else
+                    '(*args, **kwargs) -> None'
+                ),
+                My.cl: '(cls, arg2: str) -> None',
+                My().cl: '(cls, arg2: str) -> None',
+            }
+
+            bound_sigs = {
+                My().method: '(arg: int) -> None',
+                My().decorated: (
+                    '(arg: int) -> None'
+                    if follow_wrapped else
+                    '(*args, **kwargs) -> None'
+                ),
+                My.cl: '(arg2: str) -> None',
+                My().cl: '(arg2: str) -> None',
+            }
+
+            common_sigs = {
+                My.method: '(self, arg: int) -> None',
+                My.decorated: (
+                    '(self, arg: int) -> None'
+                    if follow_wrapped else
+                    '(*args, **kwargs) -> None'
+                ),
+                My.st: '(arg1: str) -> bool',
+                My().st: '(arg1: str) -> bool',
+            }
+
+            for func in (
+                inspect.signature,
+                inspect.Signature.from_callable,
+            ):
+                for fixture, text_sig in unbound_sigs.items():
```

Comment:
I'm not sure what `fixture` stands for here. signature() first parameter is called `callable`, I suggest to reuse this name here.